airflow-core/src/airflow/models/callback.py

```diff
@@ -131,26 +136,42 @@ class Callback(Base):
     trigger_id: Mapped[int] = mapped_column(Integer, ForeignKey("trigger.id"), nullable=True)
     trigger = relationship("Trigger", back_populates="callback", uselist=False)
 
-    def __init__(self, priority_weight: int = 1):
+    def __init__(self, priority_weight: int = 1, prefix: str = "", **kwargs):
         self.state = CallbackState.PENDING
         self.priority_weight = priority_weight
+        self.data = kwargs  # kwargs can be used to include additional info in metric tags
```

Comment:
Unless there are other uses I haven't gotten to yet, maybe rename `data` to `metric_tags`?